server.js

```diff
@@ -0,0 +1,111 @@
+
+var express = require('express');
+var app = express();
+var bodyParser = require('body-parser');
+app.use(bodyParser.urlencoded({ extended: true }));
+
+var path = require('path');
+app.use(express.static(path.join(__dirname, './static')));
+app.set('views', path.join(__dirname, './views'));
+app.set('view engine', 'ejs');
+
+
+var mongoose = require('mongoose');
+mongoose.connect('mongodb://localhost/giraffes');
+var GiraffeSchema = new mongoose.Schema({
+	name: String,
+	age: Number
+}, {timestamps: true});
+
+mongoose.model('Giraffe', GiraffeSchema); 
+var Giraffe = mongoose.model('Giraffe'); 
+
+
+
+// Routes
+// Shows all giraffes
+app.get('/', function(req, res) {
+	Giraffe.find({}, function(err, giraffes) {
+		if(err) {
+			console.log('could not find giraffes');
+		} else {
+			console.log('successfully found giraffes');
+			console.log(giraffes);
+			var all_giraffes = giraffes;
+
+		}
+		res.render('allgiraffes', {all_giraffes: all_giraffes});
```

Comment:
Nest your res.render into the if/else statement. How it's written right now, if you have an error, the all_giraffes variable is never declared and it will produce a syntax error.